spec/facility_spec.rb

```diff
@@ -2,25 +2,293 @@
+
+      camaro.plate_maker
+
+      expect(camaro.plate_type).to eq(:antique)
+    end
+
+    it 'makes an ev plate when vehicle is an ev engine' do
+      bolt = Vehicle.new({vin: '987654321abcdefgh', year: 2019, make: 'Chevrolet', model: 'Bolt', engine: :ev})
+
+      bolt.plate_maker
+
+      expect(bolt.plate_type).to eq(:ev)
+    end
+
+    it 'makes a regular plate when a vehicle is newer than 25 years and is not an ev engine' do
+      cruz = Vehicle.new({vin: '123456789abcdefgh', year: 2012, make: 'Chevrolet', model: 'Cruz', engine: :ice})
+
+      cruz.plate_maker
+
+      expect(cruz.plate_type).to eq(:regular)
+    end
+  end
+
+  describe '#register_vehicle' do
+    it 'starts with no registered vehicles' do
+      expect(@facility_1.registered_vehicles).to eq([])
+    end
+
+    it 'will not register a vehicle if it does not offer service' do
+      cruz = Vehicle.new({vin: '123456789abcdefgh', year: 2012, make: 'Chevrolet', model: 'Cruz', engine: :ice})
+
+      @facility_1.register_vehicle(cruz)
+
+      expect(@facility_1.registered_vehicles).to eq([])
+      expect(cruz.registration_date).to eq(nil)
+    end
+
+    it 'can keep track of a vehicle it has registered' do
+      cruz = Vehicle.new({vin: '123456789abcdefgh', year: 2012, make: 'Chevrolet', model: 'Cruz', engine: :ice})
+
+      @facility_1.add_service('Vehicle Registration')
+      @facility_1.register_vehicle(cruz)
+
+      expect(@facility_1.registered_vehicles).to eq([cruz])
+    end
+
+    it 'gives a vehicle a registration date when registered' do
+      cruz = Vehicle.new({vin: '123456789abcdefgh', year: 2012, make: 'Chevrolet', model: 'Cruz', engine: :ice})
+
+      @facility_1.add_service('Vehicle Registration')
+      @facility_1.register_vehicle(cruz)
+
+      expect(cruz.registration_date).to eq(Date.today)
+    end
+
+    it 'gives a vehicle a plate type of :regular when registered as an :ice engine' do
+      cruz = Vehicle.new({vin: '123456789abcdefgh', year: 2012, make: 'Chevrolet', model: 'Cruz', engine: :ice})
+
+      @facility_1.add_service('Vehicle Registration')
+      @facility_1.register_vehicle(cruz)
+
+      expect(cruz.plate_type).to eq(:regular)
+    end
+
+    it 'gives a vehicle a plate type of :antique when registered if vehicle is over 25 years old' do
+      camaro = Vehicle.new({vin: '1a2b3c4d5e6f', year: 1969, make: 'Chevrolet', model: 'Camaro', engine: :ice})
+
+      @facility_1.add_service('Vehicle Registration')
+      @facility_1.register_vehicle(camaro)
+
+      expect(camaro.plate_type).to eq(:antique)
+    end
+
+    it 'gives a vehicle a plate type of :ev when registered if vehicle has :ev engine:' do
+      bolt = Vehicle.new({vin: '987654321abcdefgh', year: 2019, make: 'Chevrolet', model: 'Bolt', engine: :ev})
+
+      @facility_1.add_service('Vehicle Registration')
+      @facility_1.register_vehicle(bolt)
+
+      expect(bolt.plate_type).to eq(:ev)
+    end
+  end
```

Comment:
Really great describe block for this one register_vehicle method with all the it blocks that test each of the side effects. This is great integration testing.